Mathlib/NumberTheory/NumberField/Discriminant.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2023 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.NumberTheory.NumberField.Basic
+import Mathlib.RingTheory.Localization.NormTrace
+
+/-!
+# Number field discriminant
+This file defines the discriminant of a number field.
+
+## Main definitions
+ - `discr` the discriminant of a number field.
+
+## Tags
+number field, discriminant
+-/
+
+namespace NumberField
+
+open NumberField Matrix
+
+variable (K : Type*) [Field K] [NumberField K]
+
+/-- The discriminant of a number field. -/
+noncomputable def discr : ℤ := Algebra.discr ℤ (RingOfIntegers.basis K)
+
+theorem coe_discr : (discr K : ℚ) = Algebra.discr ℚ (integralBasis K) := by
+  rw [discr]
```

Comment:
I'm not honestly sure why this is preferred, but it seems to me that having a `discr_def` theorem is often preferred instead of just `rw [discr]`.

Reply:
Another good option would be to make `discr` an `abbrev`, then you don't need the lemma at all

Reply:
> I'm not honestly sure why this is preferred, but it seems to me that having a `discr_def` theorem is often preferred instead of just `rw [discr]`. 

One problem I noticed is that simp_rw can't rewrite backwards:
```
def double (n : ℕ) : ℕ := n + n

example (n : ℕ) : double n = n + n := by
  simp_rw [← double]
```
yields
> invalid '←' modifier, 'double' is a declaration name to be unfolded

(same for `simp only` and `simp`)
while `rw [← double]`, `simp_rw [double]`, `rw [double]` all work.

Reply:
That sounds like a bug to file against `simp only`

Reply:
Note that I changed `discr` to an `abbrev`.